tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskGraph.java

```diff
@@ -640,6 +645,35 @@ private void reuseDeviceBuffersForSameDevice(TornadoDevice device) {
         }
     }
 
+    @Override
+    public void updatePersistentStates(TornadoTaskGraphInterface taskGraphSrc) {
+        TornadoTaskGraph graphSrc = (TornadoTaskGraph) taskGraphSrc;
+        List<Object> objectsToSink = executionContext.getPersistentTaskToObjectsMap()
+                .get(graphSrc.taskGraphName);
+
+        for (Object objToSink : objectsToSink) {
+            Access objectAccessSrc = graphSrc.getObjectAccess(objToSink);
+            LocalObjectState localStateSrc = graphSrc.executionContext.getLocalStateObject(objToSink, objectAccessSrc);
+            DataObjectState dataObjectStateSrc = localStateSrc.getDataObjectState();
+
+            // The device is the same for both task-graphs
+            TornadoXPUDevice device = graphSrc.meta().getXPUDevice();
```

Comment:
I think we need to add a guard to check devices are the same (origin task graph and new task graph)